api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -514,46 +518,33 @@ private FirecloudWorkspaceId generateFirecloudWorkspaceId(String namespace, Stri
     User user = userProvider.get();
-    // This code is to migrate firecloud uuid to current workspaces
-    // once we are confident this has been done we can rework/remove
-    // much of this logic
+
+    List<org.pmiops.workbench.firecloud.model.WorkspaceResponse> fcWorkspaces =
+        fireCloudService.getWorkspaces();
+    List<String> fcUuids = fcWorkspaces.stream()
+        .map(workspace -> workspace.getWorkspace().getWorkspaceId()).collect(Collectors.toList());
+
+    List<org.pmiops.workbench.db.model.Workspace> dbWorkspaces =
+        workspaceDao.findAllByFirecloudUuidIn(fcUuids);
+
     List<WorkspaceResponse> responseList = new ArrayList<WorkspaceResponse>();
-    if (user != null) {
-      List<org.pmiops.workbench.firecloud.model.WorkspaceResponse> fcWorkspaces =
-          fireCloudService.getWorkspaces();
-      for (WorkspaceUserRole userRole : user.getWorkspaceUserRoles()) {
-        org.pmiops.workbench.firecloud.model.WorkspaceResponse fcWorkspace = null;
-        org.pmiops.workbench.db.model.Workspace dbWorkspace;
-        dbWorkspace = userRole.getWorkspace();
-        List<org.pmiops.workbench.firecloud.model.WorkspaceResponse> filteredFcWorkspaces =
-            fcWorkspaces
-                .stream()
-                .filter(w -> w.getWorkspace().getName().equals(TO_CLIENT_WORKSPACE.apply(userRole.getWorkspace()).getId()))
-                .collect(Collectors.toList());
-        if (!filteredFcWorkspaces.isEmpty()) {
-          fcWorkspace = filteredFcWorkspaces.get(0);
-        }
-        if (fcWorkspace == null) {
-          log.warning(String.format(
-              "workspace '%s' workbench ACLs are out of sync with Firecloud, " +
-              "omitting this workspace for current user", dbWorkspace.getFirecloudName()));
-        } else {
-          if (dbWorkspace.getFirecloudUuid() == null) {
-            // populate UUID
-            dbWorkspace.setFirecloudUuid(fcWorkspace.getWorkspace().getWorkspaceId());
-            workspaceService.getDao().save(dbWorkspace);
-          }
-          WorkspaceResponse currentWorkspace = new WorkspaceResponse();
-          currentWorkspace.setWorkspace(TO_CLIENT_WORKSPACE.apply(dbWorkspace));
-          if (fcWorkspace.getAccessLevel().equals(WorkspaceService.PROJECT_OWNER_ACCESS_LEVEL)) {
-            currentWorkspace.setAccessLevel(WorkspaceAccessLevel.OWNER);
-          } else {
-            currentWorkspace.setAccessLevel(WorkspaceAccessLevel.fromValue(fcWorkspace.getAccessLevel()));
-          }
-          responseList.add(currentWorkspace);
-        }
+
+    for (org.pmiops.workbench.db.model.Workspace dbWorkspace : dbWorkspaces) {
+      org.pmiops.workbench.firecloud.model.WorkspaceResponse fcWorkspace = fcWorkspaces
+          .stream()
+          .filter(w -> w.getWorkspace().getWorkspaceId().equals(dbWorkspace.getFirecloudUuid()))
+          .collect(Collectors.toList())
+          .get(0);
```

Comment:
Would prefer some error checks here rather than hitting the index out of bound exception. Previous code more-or-less had this. It's definitely possible for FC workspaces and Workbench workspaces to get out of sync.